dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java

```
@@ -321,10 +322,13 @@ public void run(Timeout timeout) {
             if (future == null || future.isDone()) {
                 return;
             }
-
             ExecutorService executor = future.getExecutor();
             if (executor != null && !executor.isShutdown()) {
-                executor.execute(() -> notifyTimeout(future));
+                try {
+                    executor.execute(() -> notifyTimeout(future));
+                } catch (RejectedExecutionException e) {
+                    notifyTimeout(future);
```

I think it should throw an exception. Also, I think we need another PR to change it so that it does not call notifyTimeout, because it’s not actually a timeout.

---

Hi, just to confirm:
Thanks for your reply. Should I first change the code to throw the exception instead of catching it and calling notifyTimeout, and then submit this update in the current PR?

---

Hi @RainYuY, I've updated the code according to your feedback.
I've kept a TODO test case in place—should we leave it in, or remove it?

---

Your first version of the code had the right idea, but it missed throwing the RejectedExecutionException after the notifyTimeout call. Actually, we should not call notifyTimeout here because this is not a timeout. You can fix the notifyTimeout issue in this PR or in a follow-up PR, but for this PR please revert to your original implementation and add the exception throw.

---

OK, I get it. 
Thanks for your patience! Could you please tell me if this is correct now?